### ISSUE TITLE
Excluding popwin breaks helm

### ISSUE THREAD

**Issue by bmag** [syl20bnr/spacemacs#6634]
#### Description :octocat:
Excluding `popwin` causes error when opening helm sessions. Instead of helm session, there's an error message. As a result, helm can't be used, causing Spacemacs to be unusable.

#### Reproduction guide :beetle:
- Enable `helm` layer
- Add `popwin` to `dotspacemacs-excluded-packages`
- Start Emacs
- Wait a second for Helm to load (Spacemacs loads Helm ~1 second after startup is done)
- Start helm session: `SPC f f`

*Observed behaviour:* :eyes: :broken_heart:
Error message is printed in the echo area, Helm session does *not* open.
The error message: `Symbol’s function definition is void: popwin-mode`

With `SPC t D` (`spacemacs/toggle-debug-on-error`) before `SPC f f`, I get the backtrace posted below.

*Expected behaviour:* :heart: :smile:
Helm session opens in the same place as usual, i.e. at the bottom with a height of 0.3 of the frame height. No error.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.0.95.1
- Spacemacs: 0.105.21
- Spacemacs branch: buffer-predicate (rev. 08f5291)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm emacs-lisp)
```

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (void-function popwin-mode)
  (popwin-mode -1)
  (let ((display-buffer-base-action (quote (nil)))) (setq spacemacs-display-buffer-alist display-buffer-alist) (setq display-buffer-alist nil) (popwin-mode -1) (if helm-prevent-escaping-from-minibuffer (progn (define-key evil-motion-state-map [down-mouse-1] nil))))
  spacemacs//helm-prepare-display()
  run-hooks(helm-after-initialize-hook)
  helm-log-run-hook(helm-after-initialize-hook)
  helm-initial-setup("/home/bar/some")
  helm-initialize(nil "/home/bar/" "/home/bar/some" helm-source-find-files)
  helm-internal(helm-source-find-files "/home/bar/" "Find files or url: " nil nil "*helm find files*" nil "/home/bar/some" nil)
  apply(helm-internal (helm-source-find-files "/home/bar/" "Find files or url: " nil nil "*helm find files*" nil "/home/bar/some" nil))
  helm(helm-source-find-files "/home/bar/" "Find files or url: " nil nil "*helm find files*" nil "/home/bar/some" nil)
  apply(helm (helm-source-find-fil
```

**Comment by bmag** [comment]
Cause of the error: The function `spacemacs//helm-prepare-display` assumes `popwin` is installed and loaded. It calls `(popwin-mode -1)`, but that function is not defined because I excluded `popwin`. It's worth noting that `spacemacs//restore-previous-display-config` would also cause the same error because it calls `(popwin-mode 1)`.

I found a temporary work-around by defining `popwin-mode` in `dotspacemacs/user-init`:
```elisp
(defun popwin-mode (&rest args) (interactive) nil)
```

There's another problem with `spacemacs//helm-prepare-display` and `spacemacs//restore-previous-display-config`: if I manually disable popwin (`M-x popwin-mode`), it is automatically re-enabled after every Helm session, even if it was disabled before hand. This bothers me when I try to test something without popwin.

**Comment by syl20bnr** [comment]
The solution is to add `popwin` package to `helm` layer and move the popwin related config to `helm/post-init-popwin` or maybe `pre-init` if a `use-package` hook is required.

**Comment by bmag** [comment]
@syl20bnr ok, I'll try to make a PR before this weekend

**Comment by bmag** [comment]
A bit late, but the PR is here: #6746

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
